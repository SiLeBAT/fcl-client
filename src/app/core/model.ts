export interface ValidationError {
    code: string;
    message: string;
}
