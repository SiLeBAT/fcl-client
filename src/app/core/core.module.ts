import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {HttpClientModule, HTTP_INTERCEPTORS} from '@angular/common/http';

import {TokenInterceptor} from './services/token-interceptor.service';
import {JwtInterceptor} from './services/jwt-interceptor.service';

@NgModule({
  imports: [CommonModule, HttpClientModule],
  declarations: [],
  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: TokenInterceptor,
      multi: true,
    },
    {
      provide: HTTP_INTERCEPTORS,
      useClass: JwtInterceptor,
      multi: true,
    },
  ],
  exports: [],
})
export class CoreModule {}
