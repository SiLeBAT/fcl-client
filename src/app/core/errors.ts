import { ValidationError } from "./model";

export class InvalidServerInputHttpErrorResponse {
    constructor(
        public message: string,
        public errors: ValidationError[],
    ) {}
}
