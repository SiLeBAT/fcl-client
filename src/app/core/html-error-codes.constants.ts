export const HTML_ERROR_CODE_INTERNAL_SERVER_ERROR = 500;
export const HTML_ERROR_CODE_UNAUTHORIZED = 401;
export const HTML_ERROR_CODE_UNPROCESSABLE_ENTITY = 422;
