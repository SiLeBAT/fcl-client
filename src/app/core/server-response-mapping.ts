import { HttpErrorResponse } from "@angular/common/http";
import { HTML_ERROR_CODE_UNPROCESSABLE_ENTITY } from "./html-error-codes.constants";
import { ValidationError } from "./model";
import { SERVER_ERROR_CODE } from "./server-response.enums";
import { InvalidDataErrorDTO } from "./server-response.model";

export function isErrorWithInvalidDataErrorDTO(error: unknown): boolean {
    return (
        error instanceof HttpErrorResponse &&
        error.status === HTML_ERROR_CODE_UNPROCESSABLE_ENTITY &&
        error.error instanceof Object &&
        error.error.code === SERVER_ERROR_CODE.INVALID_INPUT &&
        error.error.errors instanceof Array &&
        error.error.errors.every(
            (valdationError: ValidationError) =>
                valdationError instanceof Object &&
                typeof valdationError.code === "string" &&
                valdationError.code !== "" &&
                typeof valdationError.message === "string" &&
                valdationError.message !== "",
        )
    );
}

export function fromErrorExtractInvalidDataErrorDTO(
    error: unknown,
): InvalidDataErrorDTO {
    if (isErrorWithInvalidDataErrorDTO(error)) {
        return (error as HttpErrorResponse).error as InvalidDataErrorDTO;
    }
    throw new Error("The InvalidDataErrorDTO is not valid.");
}
