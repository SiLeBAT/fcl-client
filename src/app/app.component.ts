import { Component, ElementRef } from '@angular/core';
import { OverlayContainer } from '@angular/cdk/overlay';

@Component({
    selector: 'fcl-root',
    templateUrl: './app.component.html'
})
export class AppComponent {

    constructor(
        private overlayContainer: OverlayContainer,
        private elementRef: ElementRef
    ) {
        // Disables all default context menus in webapp
        (this.elementRef.nativeElement as HTMLElement).oncontextmenu = e => e.preventDefault();
        this.overlayContainer.getContainerElement().oncontextmenu = e => e.preventDefault();
    }
}
