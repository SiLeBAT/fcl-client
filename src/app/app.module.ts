import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { AppRoutingModule } from './app.routing.module';
import { NgModule } from '@angular/core';
import { StoreModule } from '@ngrx/store';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { EffectsModule } from '@ngrx/effects';

import { UserModule } from './user/user.module';

import { AppComponent } from './app.component';
import { environment } from '../environments/environment';
import { MainPageModule } from './main-page/main-page.module';

@NgModule({
    declarations: [
        AppComponent
    ],
    imports: [
        BrowserModule,
        BrowserAnimationsModule,
        MainPageModule,
        UserModule,
        StoreModule.forRoot({}, {
            runtimeChecks: {
                strictStateImmutability: false,
                strictActionImmutability: false
            }
        }),
        EffectsModule.forRoot([]),
        StoreDevtoolsModule.instrument({
            name: 'FCL Devtools',
            maxAge: 25,
            logOnly: environment.production
        }),
        // AppRoutingModule needs to be at the end
        AppRoutingModule
    ],
    bootstrap: [
        AppComponent
    ]
})
export class AppModule {
}
