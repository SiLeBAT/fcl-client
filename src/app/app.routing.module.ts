import { NgModule } from "@angular/core";
import { Routes, RouterModule } from "@angular/router";

const routes: Routes = [
    // otherwise redirect to home
    { path: "**", redirectTo: "" },
];

@NgModule({
    imports: [RouterModule.forRoot(routes)],
    exports: [RouterModule],
})
export class AppRoutingModule {}
