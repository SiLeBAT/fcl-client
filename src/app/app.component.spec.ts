import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
    let fixture: ComponentFixture<AppComponent>;
    let component: AppComponent;

    beforeEach(async(() => {
        // eslint-disable-next-line @typescript-eslint/no-floating-promises
        TestBed.configureTestingModule({
            declarations: [
                AppComponent
            ],
            schemas: [ CUSTOM_ELEMENTS_SCHEMA ]
        }).compileComponents();
        fixture = TestBed.createComponent(AppComponent);
        component = fixture.debugElement.componentInstance;
    }));

    it('should create the component', async () => {
        expect(component).toBeTruthy();
    });

    it('should render correctlu', () => {
        expect(fixture).toMatchSnapshot();
    });
});
