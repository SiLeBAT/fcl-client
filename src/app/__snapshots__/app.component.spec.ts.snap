// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render correctlu 1`] = `
<fcl-root
  elementRef={[Function ElementRef]}
  overlayContainer={[Function OverlayContainer]}
>
  <fcl-main-page />
</fcl-root>
`;
